### PR TITLE
[cxx-interop] Add a basic test for using `std::iterator_traits`

### PR DESCRIPTION
**Pull request by cabmeurer**
Add basic tests for using `std::iterator_traits`

Ref: #41166

### CODE REVIEW

File: test/Interop/Cxx/stdlib/Inputs/std-iterator_traits.h [hunk 1]
```diff
@@ -0,0 +1,10 @@
+#ifndef TEST_INTEROP_CXX_STDLIB_INPUTS_STD_ITERATORTRAITS_H
+#define TEST_INTEROP_CXX_STDLIB_INPUTS_STD_ITERATORTRAITS_H
+
+#include <iterator>
+
+using IteratorTraits = std::iterator_traits<int>;
+
+inline IteratorTraits initIterator() { return {}; }
```
Comment (cabmeurer) [inline test/Interop/Cxx/stdlib/Inputs/std-iterator_traits.h:8]: idk if this is the correct way for this...

Comment (cabmeurer) [inline test/Interop/Cxx/stdlib/Inputs/std-iterator_traits.h:8]: i dont think this is the right way to do this...

**Comment by cabmeurer** [comment]
I thought I would give it a go and try to add some tests here, let me know if these is even needed and/or what I should fix

**Comment by hyp** [comment]
@swift-ci please test

**Comment by cabmeurer** [comment]
@swift-ci please test

**Comment by cabmeurer** [comment]
> **Build failed** [Swift Test OS X Platform](https://ci.swift.org/job/swift-PR-macos/31106/) Git Sha - [265b8f2](https://github.com/apple/swift/commit/265b8f229daded8609875d85d73f6fce60ac8920)

Pushed a change that should resolve this failure

**Comment by cabmeurer** [comment]
@zoecarver when you get a chance, could you merge this for me?

**Review by zoecarver — CHANGES_REQUESTED** [review]
I'm sorry, I don't think that this really makes sense to test. If we are able to construct an object, it won't be nil. It might be interesting to see if we can import `iterator_traits` (a module interface test) but honestly, I'm not even sure how valuable that would be. I don't think people will be using traits like this very often in Swift. 

What would be really helpful is some tests for `std::next` and `std::prev` (if you have time, `std::distance`, and `std::advance` might be interesting too) because I anticipate those will be used very frequently. Could you update this PR to test those instead (or make a new PR)?

I really appreciate you spending time writing tests! So, I'm sorry if this is disappointing :(

**Comment by zoecarver** [comment]
I should also mention that these tests look 100% correct, so really well done for your first PR! It's just the thing you're testing doesn't really make sense.

**Pull request closed without merging**